lib/logstash/inputs/exec.rb

```diff
@@ -124,5 +134,9 @@ def wait_until_end_of_interval(duration)
     end
   end
 
+  # convert seconds (float) to nanoseconds
```

Comment:
nit: can we avoid transitioning through a lossy format, and hold nanos as our intermediate?

Reply:
sure we can, but our over-head won't make it super precise.
forgot JRuby is no longer using `currentTimeMillis` for `Time.now` but  actually does a native call directly and has nanos -> thus it's a good and valid point - updated, thanks for the advice.